File: aspnet-in-practice/06/CS/CustomControls.ControlBuilders/ZoneBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Web.UI;
using System.Collections;

namespace CustomControls.ControlBuilders
{
	public class ZoneBuilder : ControlBuilder
	{
		public override Type GetChildControlType(string tagName, IDictionary attribs)
		{
			// more options here
			if (tagName.Equals("articles", StringComparison.InvariantCultureIgnoreCase))
				return typeof(ArticleView);
			else if (tagName.Equals("box", StringComparison.InvariantCultureIgnoreCase))
				return typeof(Box);

			return null;
		}
	}
}

File: aspnet-in-practice/06/VB/CustomControls.Web/CustomControls.ControlBuilders/ZoneBuilder.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Web.UI
Imports System.Collections

Public Class ZoneBuilder
	Inherits ControlBuilder
	Public Overrides Function GetChildControlType(ByVal tagName As String, ByVal attribs As IDictionary) As Type
		' more options here
		If tagName.Equals("articles", StringComparison.InvariantCultureIgnoreCase) Then
			Return GetType(ArticleView)
		ElseIf tagName.Equals("box", StringComparison.InvariantCultureIgnoreCase) Then
			Return GetType(Box)
		End If

		Return Nothing
	End Function
End Class